src/compiler/program.ts

```diff
@@ -513,6 +513,8 @@ namespace ts {
             }
         }
 
+        const missingFilePaths = filesByName.getKeys().filter(p => !filesByName.get(p));

```

Comment:
This will have a semantic conflict with #16724 as `filesByName` will be a `Map`, which doesn't have a `getKeys()`. You can use `mapDefined<[string, string], string>(arrayFrom(m.entries()), ([path, value]) => value === undefined ? undefined : path);`. Or add a version of `mapDefined` that works on iterators to avoid the `arrayFrom`.